### ISSUE TITLE
socket: Function not implemented

### ISSUE THREAD

**Issue by Gitii** [docker/for-win#13293]
- [ x ] I have tried with the latest version of Docker Desktop
  - [ x ] I have tried disabling enabled experimental features
  - [ x ] I have uploaded Diagnostics
  - Diagnostics ID: 485AFA80-9311-4F0E-AB0C-4B95122E1DF2/20230311221115

Docker Desktop: 4.17.0 

### Actual behavior

(happened after upgrading from 4.15) - seems to be 32bit related. Was working on previous version.

### Expected behavior

ability to use i386/debian:10

### Information
The same bug as for mac: https://github.com/docker/for-mac/issues/6689
Works with 4.15.0
Fixed for 4.16 (mac) but doesn't work under Windows (4.17.0)

### Steps to reproduce the behavior
```
docker run --platform=linux/386 -it --rm i386/debian:10
root@71dfc1d7ca5b:/# ping 8.8.8.8
ping: socket: Function not implemented
```

**Comment by Aanok** [comment]
I'm experiencing the same issue and it blocks a wide part of my activity :(

**Comment by Aanok** [comment]
> > I'm experiencing the same issue and it blocks a wide part of my activity :(
> 
> @Aanok As a workaround you can install 4.15.0 which is fairly recent. That's what I did :)

Indeed, I've also downgraded to 4.16.3 for the moment. The problem is that I'm on a workstation with remotely enforced updates and it's only a matter of time before I get the latest version forcibly pushed again (as it happened in the first place; I've learned to be very wary of Docker Desktop updates).

**Comment by patrickleboutillier** [comment]
Same bug here. We have some 32-bit binaries in a 64-bit image that can no longer get a socket(). Here is the strace output:

`57    socket(AF_INET, SOCK_STREAM, IPPROTO_IP) = -1 ENOSYS (Function not implemented)`

If I rebuild one of those binaries for 64-bit it works as expected. Downgrading to Docker Desktop 4.16.3 also fixes the issue.

**Comment by Gitii** [comment]
@Aanok Can confirm that 😀
